packages/typescript/src/outputFile.ts

```diff
@@ -16,15 +16,22 @@ export interface TypescriptSourceDescription extends Partial<SourceDescription>
 /**
  * Checks if the given OutputFile represents some code
  */
-function isCodeOutputFile(name: string): boolean {
-  return !isMapOutputFile(name) && !name.endsWith('.d.ts');
+export function isCodeOutputFile(name: string): boolean {
+  return !isMapOutputFile(name) && !isDeclarationOutputFile(name);
 }
 
 /**
  * Checks if the given OutputFile represents some source map
  */
-function isMapOutputFile(name: string): boolean {
-  return name.endsWith('.map');
+export function isMapOutputFile(name: string): boolean {
+  return name.endsWith('ts.map');
+}
```

Comment:
The reason for this change is that we only need to worry about emitting `.d.ts.map` (and equivalent `.d.cts.map` & `.d.mts.map`) files. The `.js.map` files are already being emitted by the TypeScript compiler (unless the `noEmit` option is set to `true`).